packages/driver/src/datatypes/range.ts

```diff
@@ -63,3 +63,25 @@ export class Range<
         };
   }
 }
+
+export class MultiRange<
+  T extends number | Date | LocalDate | LocalDateTime | Duration
+> {
+  constructor(private readonly _ranges: Range<T>[] = []) {}
+
+  get length() {
+    return this._ranges.length;
+  }
+
+  *[Symbol.iterator]() {
+    for (const range of this._ranges) {
+      yield range;
+    }
+  }
+
+  toJSON() {
+    return {
+      ranges: this._ranges,
```

Comment:
I think this should probably just return the array, to mirror edgedb's json representation of multiranges.

Reply:
Also probably should copy the array so you can't mutate it. (Or maybe this should be done in the constructor? or both?)

Reply:
Cool, I think I just copied this from the python impl, but that makes sense!